src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -225,7 +225,11 @@ public static ZipArchive ReadZipFile (string filename, bool strictConsistencyChe
 				files.Add (outfile);
 				var dt = File.Exists (outfile) ? File.GetLastWriteTimeUtc (outfile) : DateTime.MinValue;
 				if (forceUpdate || entry.ModificationTime > dt) {
-					entry.Extract (destination, fullName, FileMode.OpenOrCreate);
+					try {
+						entry.Extract (destination, fullName, FileMode.OpenOrCreate);
+					} catch (PathTooLongException) {
+						throw new PathTooLongException ($"Could not extract {fullName} to {outfile}. Path is too long.");
```

Comment:
Please "quote" names, so that they're more easily understandable when the contain spaces, e.g.:

    Could not extract `{fullName}` to `{outfile}`.